components/coding/studentPortal/Layout.tsx

```diff
@@ -1,10 +1,37 @@
+import { useQuery } from "@apollo/client";
+import { format, addDays } from "date-fns";
 import React, { useState } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import {
+  FetchGoalCountResponse,
+  FETCH_USER_GOALS_COUNT,
+} from "../../../graphql/fetchUserGoalsCount";
+import { useAuth } from "../../../lib/authContext";
+import {
+  isGoalApproaching,
+  userGoalsSelector,
+} from "../../../redux/userGoalsSlice";
 import Navbar from "../../ui/Navbar";
 import Sidebar from "./Sidebar";
 
 export const Layout: React.FC = ({ children }) => {
   const [active, setActive] = useState(false);
 
+  const { user } = useAuth();
+  const goalDateThreshold = format(addDays(new Date(), 7), "MM/dd/yyyy");
+  const goalApproaching = useSelector(userGoalsSelector);
+  const dispatch = useDispatch();
+  const { data } = useQuery<FetchGoalCountResponse>(FETCH_USER_GOALS_COUNT, {
+    variables: {
+      userId: user.uid,
+      goalDateThreshold: goalDateThreshold,
+    },
+    onCompleted: (data: FetchGoalCountResponse) => {
+      if (data.user_goals_aggregate != null) {
+        dispatch(isGoalApproaching(goalApproaching));
```

Comment:
You're not using the data coming back from the query? You're just dispatching an action with the existing value in the selector?

Reply:
I think the kind of logic you're looking for here is:
- does the FETCH_USER_GOALS_COUNT query return a response i.e is there any upcoming goals? (you're doing this)
- if so, you want to dispatch an action to set goalApproaching to true

the way i'm reading your logic right now is: set the goalApproaching value to the value of goalApproaching (i.e never change it)

Reply:
fixed.